### PR TITLE
dump all signals to waveform for VCS 

### PR DESCRIPTION
**Pull request by ni140**
Create simple TCL file that is passed with "-do" parameter same manner that's done for Questa.
This allows all signals to be properly logged on waveform after test finishes

### CODE REVIEW

**Comment by ni140** [comment]
Can someone please merge this?

File: cocotb/share/makefiles/simulators/Makefile.vcs [hunk 1]
```diff
@@ -73,11 +73,19 @@ $(SIM_BUILD)/simv: $(SIM_BUILD) $(VERILOG_SOURCES) $(SIM_BUILD)/pli.tab $(COCOTB
 	LD_LIBRARY_PATH=$(LIB_DIR):$(LD_LIBRARY_PATH) TOPLEVEL=$(TOPLEVEL) \
 	$(CMD) -top $(TOPLEVEL) $(PLUSARGS) +acc+1 +vpi -P pli.tab +define+COCOTB_SIM=1 -sverilog $(EXTRA_ARGS) -debug -load libvpi.so $(COMPILE_ARGS) $(VERILOG_SOURCES)
 
+$(SIM_BUILD)/runsim.tcl:
+	@echo "# Autogenerated file" > $@
+ifeq ($(GUI),1)
+	@echo "dump -aggregates -add /" >> $@
+	@echo "run" >> $@
+	@echo "quit" >> $@
+endif
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.vcs]: Why do you only dump signals in gui mode?

Comment (ni140) [reply cocotb/share/makefiles/simulators/Makefile.vcs]: otherwise there won't be any signal values visible on waveform

**Comment by imphil** [comment]
Thanks for this patch, and sorry for the wait.

- Is it possible to specify "-do" twice in VCS? Aldec and Questa use SCRIPT_FILE to work around that, this must be implemented then as well.
- Mentor and Riviera make the waveform dumping depend on the WAVES variable (https://cocotb.readthedocs.io/en/latest/building.html#var-WAVES). Can you do the same?
- Please also update the documentation at https://cocotb.readthedocs.io/en/latest/building.html#var-WAVES to mention VCS.

Overall, I'm not convinced it is worth it. I'm actually more inclined to remove this functionality from Riviera Pro and Questa and telling users to use EXTRA_ARGS for use cases like that, and not expose more and more abstracted simulator functionality through cocotb.

**Comment by imphil** [comment]
Thanks for your contribution. Unfortunately, this PR has not been updated to include the requested changes. I'm therefore closing this PR for now. Please re-open it if you want to continue working on this feature.

**Pull request closed without merging**